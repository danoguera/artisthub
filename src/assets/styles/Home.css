.mainContainer h1{
    font-size: 40px;
    text-align: center;
    color: white;
  }

  .mainContainer{
    padding: 20px 0;
  }
  

.services{
    width: 100%;
   }
   
   .services div{
     margin: 10px 16px 0;
     width: calc(100% - 32px);
     height: 400px;
   }
   
   .service1{
     background-image: url("../images/mujerfoto.jpeg");
     background-repeat: no-repeat;
     background-position: center;
     background-size: cover;
     color: #FFFFFF;
     opacity: 0.8;
   }
   
   .service1 h3{
     font-size: 50px;
     color: #FFFFFF;
     text-align: center;
     text-shadow: 2px 2px 8px #111;
     font-weight: 900;
     cursor: pointer;
     line-height: 370px;
   }
   
   .service2{
     background-image: url("../images/camfoto.jpeg");
     background-repeat: no-repeat;
     background-position: center;
     background-size: cover;
     color: #FFFFFF;
     opacity: 0.9;
   }
   
   .service2 h3{
     font-size: 50px;
     color: #FFFFFF;
     text-align: center;
     text-shadow: 2px 2px 8px #111;
     font-weight: 900;
     line-height: 370px;
   }
   
   .service3{
     background-image: url("../images/micfoto.jpeg");
     background-repeat: no-repeat;
     background-position: center;
     background-size: cover;
     color: #FFFFFF;
     opacity: 0.9;
   }
   
   .service3 h3{
     font-size: 50px;
     color: #FFFFFF;
     text-align: center;
     text-shadow: 2px 2px 8px #111;
     font-weight: 900;
     line-height: 370px;
   }

   @media only screen and (min-width: 1000px){

    .services div{
        margin: 16px 5px 0 28px;
        width: calc(33% - 35px);
        height: 400px;
        float: left;
      }
    
      .service2 h3{
        line-height: 280px;
      }
    
      .service1 h3{
        line-height: 280px;
      }
    
      .service3 h3{
        line-height: 280px;
      }

   }
   