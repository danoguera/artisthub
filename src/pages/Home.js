import React from 'react';
import '../assets/styles/Home.css'
import {Link} from 'react-router-dom';

class Home extends React.Component {
    constructor() {
        super();
        this.state = {

        }
    }

    // Esto ya se puede borrar?
    // handleCategory = (event) => {
    //     if (String(event.target).indexOf("Video")>0){
    //         this.props.history.push("/category/music");   
    //     }
    //     if (String(event.target).indexOf("Video")>0){
    //         this.props.history.push("/category/videography");   
    //     }
  
    // } 

    handlePhoto = (event) => {
        this.props.history.push("/photographers");
    }

    render() {
        return (

            <React.Fragment>
                <div className="mainContainer">
                    <h1>Tell us which service you are looking for:</h1>

                    <div className="services">
                        <div  onClick={ this.handlePhoto } data-testid="photography-view" className="service1">
                            <h3>Photo</h3>
                        </div>
                        <div  className="service2">
                            <h3><Link to="/category/Videography" className="text-link" >Video</Link></h3>
                        </div>      
                        <div className="service3">
                            <h3><Link to="/category/Music" className="text-link" >Music</Link></h3>
                        </div>
                    </div>
                </div>
            </React.Fragment>
        )
    }
}

export default Home;