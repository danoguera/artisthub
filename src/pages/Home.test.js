import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { MemoryRouter, Router} from 'react-router-dom';
import { createMemoryHistory } from "history";
import Home from './Home';

describe('Home', () => {

  afterEach(() => {
    cleanup();
  });

  it('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should send the user to photographers posts when button is clicked', (done) => {
    const history = [];   //Toca pasar este props
    const { getByTestId } = render(
        <MemoryRouter>
            <Home history={history} />
        </MemoryRouter>
    );

      fireEvent.click(getByTestId('photography-view'));
      expect(history[0]).toMatch("/photographers"); 
      done();

   });

   

   // check that the content changed to the new page
   //expect(container.innerHTML).toMatch('You are on the about page')

   it('should jump to Video page page when button is clicked', (done) => {
    const history = createMemoryHistory();
    const { getByText } = render(
        <Router history={history} >
            <Home/>
        </Router>
    );

      fireEvent.click(getByText(/Video/i))
      expect(history.location.pathname).toMatch("/category/Videography"); 
      done();

   });

   it('should jump to Music page page when button is clicked', (done) => {
    const history = createMemoryHistory();
    const { getByText } = render(
        <Router history={history} >
            <Home/>
        </Router>
    );

      fireEvent.click(getByText(/Music/i))
      expect(history.location.pathname).toMatch("/category/Music"); 
      done();

   });

})