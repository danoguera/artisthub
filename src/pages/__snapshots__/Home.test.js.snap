// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div
  class="mainContainer"
>
  <h1>
    Tell us which service you are looking for:
  </h1>
  <div
    class="services"
  >
    <div
      class="service1"
      data-testid="photography-view"
    >
      <h3>
        Photo
      </h3>
    </div>
    <div
      class="service2"
    >
      <h3>
        <a
          class="text-link"
          href="/category/Videography"
        >
          Video
        </a>
      </h3>
    </div>
    <div
      class="service3"
    >
      <h3>
        <a
          class="text-link"
          href="/category/Music"
        >
          Music
        </a>
      </h3>
    </div>
  </div>
</div>
`;
